src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Deserialization.kt

```diff
@@ -64,30 +66,54 @@ object Deserialization {
         lastSent = readLightningMessage() as ChannelReady
     )
 
+    private fun Input.readWaitForFundingSignedWithPushAmount(): WaitForFundingSigned {
```

Comment:
Nit: the ordering is weird, I think it should be:
1. `readWaitForFundingSigned` 
2. `readWaitForFundingSignedWithPushAmount` 
3. `readWaitForFundingSignedLegacy`